### PR TITLE
Kc docker

### PR DESCRIPTION
**Pull request by ciebson123**
Added Dockerfiles and compose

Additionally added dummy fastapi aplications, because we don't have a proper server yet

### CODE REVIEW

**Comment by AaRaBiNoZa** [comment]
I only took a quick peek for now -- will complete a review a little later, but I want to ask: do we want to keep the mapper and reducer split even here? I thought we expected to have a general "worker" whose same instance can handle both map and reduce tasks, in other words, just have a single rpc server with two "servicers" for map/reduce tasks.

**Comment by ciebson123** [comment]
> I only took a quick peek for now -- will complete a review a little later, but I want to ask: do we want to keep the mapper and reducer split even here? I thought we expected to have a general "worker" whose same instance can handle both map and reduce tasks, in other words, just have a single rpc server with two "servicers" for map/reduce tasks.

True, switched to master/worker without mapper/reducer split. We don't want to split those

File: docker/worker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+FROM python:${PYTHON_VERSION}-slim AS base
+
+# Prevents Python from writing pyc files.
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Keeps Python from buffering stdout and stderr to avoid situations where
+# the application crashes without emitting any logs due to buffering.
+ENV PYTHONUNBUFFERED=1
+
+WORKDIR /mapper
```
Comment (AaRaBiNoZa) [inline docker/worker/Dockerfile]:
```suggestion
WORKDIR /worker
```

File: docker/worker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+RUN ./build_protos.sh
+
+# Switch to the non-privileged user to run the application.
+USER appuser
+
+# Expose the port that the application listens on.
+EXPOSE 8000
+
+# Run the application.
+CMD python3 -m uvicorn docker.mapper.example_worker:app --host=0.0.0.0 --port=8000
```
Comment (AaRaBiNoZa) [inline docker/worker/Dockerfile]:
```suggestion
CMD python3 -m uvicorn docker.worker.example_worker:app --host=0.0.0.0 --port=8000
```

**Review by AaRaBiNoZa — APPROVED** [review]
Left two changes, probably left from a previous version. Without it the container crashes.
Otherwise, checked it locally and lgtm, if you only change those 2 lines consider it approved and ready to merge.

**Pull request merged**